sympy/physics/mechanics/joint.py

```diff
@@ -474,6 +474,55 @@ def _locate_joint_frame(self, body, interframe):
         body.masscenter.set_vel(interframe, 0)  # Fixate interframe to body
         return interframe
 
+    def _fill_coordinate_list(self, coordinates, n_coords, label='q', offset=0,
+                              number_single=False):
+        """Helper method for _generate_coordinates and _generate_speeds.
+
+        Parameters
+        ==========
+
+        coordinates : iterable
+            Iterable of coordinates or speeds that have been provided.
+        n_coords : Integer
+            Number of coordinates that should be returned.
+        label : String, optional
+            Coordinate type either 'q' (coordinates) or 'u' (speeds). The
+            Default is 'q'.
+        offset : Integer
+            Count offset when creating new dynamicsymbols. The default is 0.
+        number_single : Boolean
+            Boolean whether if n_coords == 1, number should still be used. The
+            default is False.
+
+        """
+
+        def create_symbol(number):
+            if n_coords == 1 and not number_single:
+                return dynamicsymbols(f'{label}_{self.name}')
+            return dynamicsymbols(f'{label}{number}_{self.name}')
+
+        name = 'generalized coordinate' if label == 'q' else 'generalized speed'
+        coords = []
```

Comment:
It's slightly tricky to follow the logic of this function with `coords` and `coordinates`. Can one of them be given a more descriptive name? For example, make `coords` something like `new_coords` or `generated_coordinates`.

Reply:
This is part of #24028

Reply:
[It](https://github.com/sympy/sympy/pull/24028/commits/7ac3d6e2dbbbc41a9971031aef3dedebb5378d4a) has been. I changed it to `generated_coordinates`, since that is most clear and does not require wrapping. I'll be updating this branch to use the latest commits of the master.